### ISSUE TITLE
[java] ArrayIsStoredDirectly false negative when parenthesized

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3933]
**Affects PMD Version:** 
6.42.0

**Rule:** [ArrayIsStoredDirectly](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#arrayisstoreddirectly)

**Description:**
Hi, I found a false negative about the rule ArrayIsStoredDirectly. PMD should have reported a warning in line 3 because the array is stored directly. Actually, I think this false negative is caused by a "complex" expression `(x)`. Thanks for your consideration.

**Code Sample demonstrating the issue:**

```java
String [] arr;
void foo (String[] x) {
    arr = (x);  // should report a warning here
}
```

**Expected outcome:**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by Zustin** [comment]
I found the rule [IdempotentOperations](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#idempotentoperations) also has a similar problem. For the following exmaple, we should report a warning in line 3.
```java
public class Foo {
  private void bar() {
      x = (x);  // should report a warning here
  }
}
```

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
